wordle-web/pages/wordlist.vue

```diff
@@ -0,0 +1,151 @@
+            label="Search Words"
+            @input="updateList()"
+          >
+          </v-text-field>
+        </v-row>
+        <v-row>
+          <v-col :cols="9">
+            <v-text-field v-model="newWord" label="New Word"> </v-text-field>
+          </v-col>
+          <v-col>
+            <v-btn @click="addWord"> Add Word </v-btn>
+          </v-col>
+        </v-row>
+        <v-simple-table>
+          <thead>
+            <tr>
+              <th>Word</th>
+              <th style="text-align: center">Common Word</th>
+              <th style="text-align: center">Delete?</th>
+            </tr>
+          </thead>
+          <tbody>
+            <tr v-for="(word, index) in words" :key="index">
+              <td>{{ word.value }}</td>
+              <td style="text-align: center">
+                <v-checkbox
+                  v-model="word.common"
+                  @click="toggleCommonWord(word.value, word.common)"
+                ></v-checkbox>
+              </td>
+              <td style="text-align: center">
+                <v-btn color="primary" @click="deleteWord(word.value)">
+                  Delete
+                </v-btn>
+              </td>
+            </tr>
+          </tbody>
+        </v-simple-table>
+      </v-card-text>
+      <template>
+        <div>
+          <v-pagination
+            v-model="page"
+            circle
+            :length="numberOfPages"
+            @input="updateList()"
+          >
+          </v-pagination>
+        </div>
+      </template>
+    </v-card>
+  </v-container>
+</template>
+
+<script lang="ts">
+import { Component, Vue } from 'vue-property-decorator'
+
+@Component({})
+export default class wordlist extends Vue {
+  words: any = []
+  listSize = 1
+  wordPerPage = 50
+  numberOfPages = 10
+  newWord = ''
+  page = 1
+  search = ''
+
+  mounted() {
+    this.updateList()
+  }
+
+  addWord() {
+    this.$axios
+      .post('/Word/AddWord', {
+        value: this.newWord,
+        common: false,
+      })
+      .then(this.updateList)
+  }
+
+  deleteWord(word: string) {
+    this.$axios
+      .post('/Word/RemoveWord', null, {
+        params: { word },
+      })
+      .then(this.updateList)
+  }
+
+  toggleCommonWord(word: string, common: boolean) {
+    this.$axios.post('/Word/ChangeFlag', {
+      value: word,
+      common,
+    })
+  }
+
+  updateList() {
+    if (this.search === '') {
+      this.$axios
+        .get('/Word/GetList', {
+          params: {
+            pageNum: this.page,
+            pageSize: this.wordPerPage,
+          },
+        })
+        .then((res) => (this.words = res.data))
+        .then(this.getListSize)
+    } else {
+      this.$axios
+        .get('/Word/GetFilteredList', {
+          params: {
+            pageNum: this.page,
+            pageSize: this.wordPerPage,
+            filter: this.search,
+          },
+        })
+        .then((res) => (this.words = res.data))
+        .then(this.getListSize)
+    }
```

Comment:
Minor ux issue when you search for words. The page is set to be the current page after retrieving filtered words, which can result in a query appearing to deliver no results if you are on, say, page 4 but the query only produces two pages of results. Would recommend setting the page number to 1 after updating the list.